src/components/files/FilePreview/SpreadsheetView.tsx

```diff
@@ -76,16 +103,22 @@ const RenderedSheet: FC<Omit<SheetData, 'name'>> = (props) => {
             return <th key={key}>{name}</th>;
           })}
         </tr>
-        {rows.map((row, index) => (
-          <tr key={index}>
-            <td key={index} className="table-header">
-              {index}
-            </td>
-            {columns.map((column) => (
-              <td key={column.key}>{row[column.key]}</td>
-            ))}
-          </tr>
-        ))}
+        {rows.map((cells, rowIndex) => {
+          return (
+            <tr key={rowIndex}>
+              <th className="table-header">{rowIndex + 1}</th>
+              {cells.map((cell) => {
+                if (cell.spanned) return null;
```

Comment:
Trying to follow logic...are you padding empty cells in parsing above but then just ignoring them here?
Or am I missing another reason for that?

Reply:
Correct. It's HTML, so the way you render a spanned `<td>` is to not render it

Reply:
Okay, now that my brain has revved up a little I realize that you were probably asking why we're bothering to keep the `spanned: true` cells around at all. It's because the same item in the array of arrays that makes up the row data returned by `sheet_to_json` might need to be marked as `spanned` by more than one cell-merge. If we just removed it from the list when processing "Merge A", but then looked for its index again later when processing "Merge B", we'd find a totally different item and mark that one as `spanned` incorrectly.

Reply:
Ok got it. And it's important to look it up by index? Or more specifically is it necessary to map to an object? If we were processing the actual data here I would say yes. But since we are just displaying via two loops we don't care about those keys/indexes. Or...?

Reply:
`sheet_to_json` gives us something like this:

```
rows: [
  ["A1", B2", C2-D2", ""],
  ["A2", "B2-B3", "C2, D2"],
  ["A3", "", "C3-D4", ""],
  ["A4", "B4", "", ""],
]
```

If all we were dealing with were merged columns, we wouldn't need to use objects for tracking the state of our processing and the resulting `rowspan`/`colspan` values. We could just pass `SpreadsheetView` the raw `sheet_to_json` data, and then in our `.map()` just always look ahead of each current index to see if the next one or more items are blank, and if so, set `colspan` to the number of blank items + 1. But trying to do that look-ahead operation for subsequent rows, particularly when more than one column might be merged along with the rows, seems like it would get very messy. It seemed like a better idea to just process all the merging data up front and then just pass the "view" component something it could render pretty simply.

Reply:
(Also, when I referred to looking up a cell by "index" above, I meant not the `index` value in the object we're creating here but the index its array. What I'm saying is we can't just delete empty strings from the row-data arrays and have things turn out the same.)

Also, an empty string in the row data isn't the same as a merge. It might literally be an empty cell. We need a way to distinguish, which is why we're using the SheetJS `worksheet['!merges']` value to get the list of merged cells and then tag the values in the row data accordingly.

Reply:
I gotcha. In other words all of this info is stored in separate lists instead of a tree structure. Then yes, I'd agree that that processing/merging should be done separately like you've done.